lib/utils.js

```diff
@@ -174,7 +174,19 @@ function checkRpId(rpId) {
 
 	var isLocalhost = (rpId === "localhost" || rpId.endsWith(".localhost"));
 
-	if (!psl.isValid(rpId) && !isLocalhost) {
+	var rpIdUrl = null;
+	var isValidUrl = true;
+	try {
+		rpIdUrl = new URL(rpId);
```

Comment:
as mentioned in https://github.com/webauthn-open-source/fido2-lib/pull/63/files#r687515647 psl.isValid looks for a domain and not full url, and in webauth we have two possible rpId values:
- full domain - default webauthn case (e.g. example.com)
- full url - case of u2f migrated webauthn credentials (e.g. https://www.example.com)

Due to this we might have to check against two possible values, domain and url. So the isValidUrl simply tries to verify if rpId is a valid url, in case of full domain (default flow) it would be false but since full domain is also allowed psl.isValid would return true and value would be accepted. So basically conditions should be at least one of: isValidDomain, isLocalhost, isValidUrl

Reply:
In master branch flow if we try to pass a full url in rpId it would throw `rpId is not a valid eTLD+1` error